### PR TITLE
Warn when there are excessive duplicate jobs registered

### LINKED ISSUE #7: Cron check should alert if too many tasks are scheduled two close together
Sometimes badly coded plugins can lead to dozens of instances of the same task to be scheduled at the same time or within seconds of each other. This check should look for a pattern and alert if it exists.

https://github.com/pantheon-systems/wp_launch_check/edit/master/CHECKS.md#71

### PR DESCRIPTION
**Pull request by danielbachhuber**
Fixes #7

### CODE REVIEW

**Comment by danielbachhuber** [comment]
@stevector #review (I'll have to merge because you don't yet have contributor access on this repo, cc @joshkoenig)

File: features/cron.feature [hunk 1]
```diff
@@ -58,5 +62,9 @@ Feature: Check crons
     When I run `wp launchcheck cron`
     Then STDOUT should contain:
       """
+      Some jobs are registered more than 10 times, which is excessive and may indicate a problem with your code.
```
Comment (stevector) [inline features/cron.feature]: @danielbachhuber any reason not to have the `These jobs include: ' . implode( ', ', $excessive_jobs ),` part of the message in the test?

**Comment by stevector** [comment]
@danielbachhuber, looking good. I wonder about the `checkCron()` method. It now seems to be four checks in one. Is there a number of check or total number of lines in the method at which you would want to break this method apart?

File: features/cron.feature (hunk 1 above)
Comment (danielbachhuber) [inline features/cron.feature]: No reason not to. I've added in 60c2e3e

**Comment by danielbachhuber** [comment]
> I wonder about the checkCron() method. It now seems to be four checks in one. Is there a number of check or total number of lines in the method at which you would want to break this method apart?

Personally, I'm not very particular about the length of a method. If a given block of code is only used once, it doesn't make any functional difference to have the code as a part of a larger method (which is easier to read) or broken into private methods. If a given block of code needs to be reused, then it clearly makes sense to break it into a separate method.

In this case, I'm more comfortable shimming a new check in the exist code path than refactoring the entire class.

**Comment by stevector** [comment]
Yeah, I agree that there is not a functional difference with code used once. I personally differ on the readability question. I think many small functions are easier to read than one big one. But I suppose that is personal preference. Let's merge.

**Pull request merged**